roles/openshift_cli/library/container_binary_sync.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# vim: expandtab:tabstop=4:shiftwidth=4
+# pylint: disable=missing-docstring
+#
+
+import random
+import tempfile
+import shutil
+import subprocess
+import os.path
+
+from ansible.module_utils.basic import AnsibleModule
+
+
+DOCUMENTATION = '''
+---
+module: container_binary_sync
+short_description: Copies OpenShift binaries out of the given image tag to host system.
+'''
+
+
+class BinarySyncError(Exception):
+    def __init__(self, msg, cmd, exit_code, output):
+        super(BinarySyncError, self).__init__(msg)
+        self.msg = msg
+        self.cmd = cmd
+        self.exit_code = exit_code
+        self.output = output
+
+
+def run_cmd(args):
+    proc = subprocess.Popen(args, stdout=subprocess.PIPE)
+    output = proc.stdout.read()
+    return output
+
+
+# pylint: disable=too-few-public-methods
+class BinarySyncer(object):
+    """
+    Syncs the openshift, oc, oadm, and kubectl binaries/symlinks out of
+    a container onto the host system.
+    """
+
+    def __init__(self, image, tag):
+        self.changed = False
+        self.output = []
+        self.bin_dir = '/usr/local/bin'
+        self.image = image
+        self.tag = tag
+        self.temp_dir = None # TBD
+
+    def sync(self):
+        container_name = "openshift-cli-%s" % random.randint(1, 100000)
+        self.output.append(run_cmd(['docker', 'create', '--name',
```

Comment:
Good point, but if we plan to leverage the docker modules for ensuring images for containerized at some point, we will need the dependency anyway.

Adding the dep now or then doesn't really make a big difference to me.